internal/keepers/util.go

```diff
@@ -158,7 +158,12 @@ func observationsToUpkeepKeys(logger *log.Logger, observations []types.Attribute
 
 		// if we have a non-empty list of upkeep identifiers, limit the upkeeps we take to observationUpkeepsLimit
 		if len(upkeepObservation.UpkeepIdentifiers) > 0 {
```

Comment:
ah good catch. Can you also add a unit test for the behaviour that was untested before which caused panic. (sorry should have pointed that out earlier but let's add that in a followup PR)

Reply:
Not really. We're using a constant to set length limit. The constant would need to be changed to a configuration variable instead. The issue brought up was that IF that constant ever changed, it would panic.